src/gen/lib/Codegen_boilerplate.ml

```diff
@@ -0,0 +1,120 @@
+(*
+   Derive boilerplate to be used as a template when mapping a CST
+   to another type of tree.
+*)
+
+open Printf
+open CST_grammar
+open Indent.Types
+
+let trans = Codegen_util.translate_ident
+
+let make_header grammar = sprintf
+{|(**
+   Boilerplate to be used as a template when mapping the %s CST
+   to another type of tree.
+*)
+
+(* Disable warnings against unused variables *)
+[@@@warning "-26-27"]
+
+(* Disable warning against unused 'rec' *)
+[@@@warning "-39"]
+
+let token (_tok : Tree_sitter_run.Token.t) = failwith "not implemented"
+let blank () = failwith "not implemented"
+let todo _ = failwith "not implemented"
+
+|}
+    grammar.name
+
+let todo = [ Line "todo ()" ]
+
+let simple_pat = function
+  | Symbol _ -> "x"
+  | Token _ -> "tok"
+  | Blank -> "()"
+  | Repeat _ -> "xs"
+  | Repeat1 _ -> "xs"
+  | Choice _ -> "x"
+  | Optional _ -> "opt"
+  | Seq l ->
+      sprintf "(%s)"
+        (List.mapi (fun i _ -> sprintf "v%i" (i+1)) l
+         |> String.concat ", ")
+
+let simple_exp x = simple_pat x
+
+let gen_rule_mapper_body body =
+  let arg = simple_exp body in
+  match body with
+  | Symbol name -> [ Line (sprintf "map_%s %s" (trans name) arg) ]
```

Comment:
I would remove the map_, we know the whole file is about a conversion function. See my js_to_generic.ml,
python_to_generic.ml.